src/HucaresServer.Storage.UnitTests/MissingLicensePlateTests.cs

```diff
@@ -0,0 +1,213 @@
+            var result = missingLicensePlateHelper.InsertPlateRecord(expectedPlate, expectedDate);
+
+            //Assert
+            A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
+                .MustHaveHappenedOnceExactly();
+
+            A.CallTo(() => fakeDbSet.Add(result))
+                .MustHaveHappened();
+
+            A.CallTo(() => fakeHucaresContext.SaveChanges())
+                .MustHaveHappenedOnceExactly();
+
+            result.PlateNumber.ShouldBe(expectedPlate);
+            result.SearchStartDateTime.ShouldBe(expectedDate);
+        }
+
+        [TestMethod]
+        public void InsertMissingPlate_WhenPlateNumberIsNotValid_ShouldThrowAnError()
+        {
+            //Arrange
+            var fakeDbSet = A.Fake<DbSet<MissingLicensePlate>>();
+
+            var fakeHucaresContext = A.Fake<HucaresContext>();
+            A.CallTo(() => fakeHucaresContext.MissingLicensePlates)
+                .Returns(fakeDbSet);
+
+            var fakeDbContextFactory = A.Fake<IDbContextFactory>();
+            A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
+                .Returns(fakeHucaresContext);
+
+            var missingPlateHelper = new MissingPlateHelper(fakeDbContextFactory);
+
+            var dateInput = "Jan 1, 2018";
+            DateTime expectedDate = DateTime.Parse(dateInput);
+            
+            //Act & Assert
+            Assert.ThrowsException<UriFormatException>(() => missingPlateHelper.InsertPlateRecord("5555555", expectedDate));
+
+            A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
+                .MustNotHaveHappened();
+        }
+
+        [TestMethod]
+        public void GetAllPlates_WhenDbIsEmpty_ShouldThrowEmptyDB()
+        {
+            //Arrange
+            var missingPlateObj = new MissingLicensePlate();
+            var fakeIQueryable = new List<MissingLicensePlate>()
+            {
+                missingPlateObj,
+                new MissingLicensePlate()
+            }.AsQueryable();
+
+            var fakeDbSet = StorageTestsUtil.SetupFakeDbSet(fakeIQueryable);
+
+            var fakeHucaresContext = A.Fake<HucaresContext>();
+            A.CallTo(() => fakeHucaresContext.MissingLicensePlates)
+                .Returns(fakeDbSet);
+
+            var fakeDbContextFactory = A.Fake<IDbContextFactory>();
+            A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
+                .Returns(fakeHucaresContext);
+
+            var missingPlateHelper = new MissingPlateHelper(fakeDbContextFactory);
+
+            //Act
+            var result = missingPlateHelper.GetAllPlateRecords();
+
+            //Assert
+            A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
+                .MustHaveHappenedOnceExactly();
+
+            A.CallTo(() => fakeHucaresContext.SaveChanges())
+                .MustNotHaveHappened();
+
+            result.Count().ShouldBe(0);
+            result.FirstOrDefault().ShouldBe(missingPlateObj);
```

Comment:
If you expect the count to be 0, then FirstOrDefault will always be null. In theory this will make a null reference exception.